crates/core/src/storage/mod.rs

```diff
@@ -30,6 +38,294 @@ lazy_static! {
     static ref DELTA_LOG_PATH: Path = Path::from("_delta_log");
 }
 
+/// Creates static IO Runtime with optional configuration
+fn io_rt(config: Option<&RuntimeConfig>) -> &'static Runtime {
+    static IO_RT: OnceLock<Runtime> = OnceLock::new();
+    IO_RT.get_or_init(|| {
+        let rt = match config {
+            Some(config) => {
+                let mut builder = if config.multi_threaded {
+                    RuntimeBuilder::new_multi_thread()
+                } else {
+                    RuntimeBuilder::new_current_thread()
+                };
+                let builder = builder.worker_threads(config.worker_threads);
+                let builder = if config.enable_io && config.enable_time {
+                    builder.enable_all()
+                } else if config.enable_io && !config.enable_time {
+                    builder.enable_io()
+                } else if !config.enable_io && config.enable_time {
+                    builder.enable_time()
+                } else {
+                    builder
+                };
+                builder
+                    .thread_name(
+                        config
+                            .thread_name
+                            .clone()
+                            .unwrap_or("IO-runtime".to_string()),
+                    )
+                    .build()
+            }
+            _ => Runtime::new(),
+        };
+        rt.expect("Failed to create a tokio runtime for IO.")
+    })
+}
+
+/// Configuration for Tokio runtime
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct RuntimeConfig {
+    multi_threaded: bool,
+    worker_threads: usize,
+    thread_name: Option<String>,
+    enable_io: bool,
+    enable_time: bool,
+}
+
+/// Provide custom Tokio RT or a runtime config
+#[derive(Debug, Clone)]
+pub enum IORuntime {
+    /// Tokio RT handle
+    RT(&'static Runtime),
```

Comment:
instead of an instance of borrowed Runtime you could accept an owned Handle to the runtime which one could get with 

`runtime.handle().clone()`

Reply:
Actually that's what I did first but it didn't work. All tasks got canceled

Reply:
Donee now